packages/compiler/core/semantic-walker.ts

```diff
@@ -238,6 +238,30 @@ function navigateType(
   }
 }
 
+// Find the immediate children of the given model.
+export function getChildModels(program: Program, parent: ModelType) {
+  const children: ModelType[] = [];
+  navigateProgram(program, {
+    model: (model) => {
+      if (model.baseModel === parent) {
+        children.push(model);
+      }
+    },
+  });
+  return children;
+}
+
+// Return property from type, nesting into baseTypes as needed.
+export function getProperty(type: ModelType, propertyName: string): ModelTypeProperty | undefined {
+  if (type.properties.has(propertyName)) {
+    return type.properties.get(propertyName);
+  }
+  if (type.baseModel) {
+    return getProperty(type.baseModel, propertyName);
```

Comment:
Consider converting this tail call into a loop since node JS engine doesn't do tail call optimization.

Reply:
done.